### ISSUE TITLE
Unify Callback Function Naming

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go#966]
[From recent feedback](https://docs.google.com/document/d/1WQ9_KQTAu2cfkh4A0OZFB9DfEtZkBa-1x10JuQJK5mM/edit):

> Currently, there isn’t a convention in naming for callback-like functions. `GetHookFunc`, `BatchObserverCallback`. This gives users an impression there is a specific reason they are named differently and may have differences in behavior although they don’t.
> We can pick one for consistency and developer productivity. `*Callback` is used less in Go, `*Func` or `*Handler` would be a good fit.

**Comment by dengliming** [comment]
@MrAlias Hello, I'd love to take this as first contribution for this repo. I suggest rename `*Callback` with `*Handler`. WDYT?

**Comment by MrAlias** [comment]
Thanks for the help. I'll assign this issue to you.

> @MrAlias Hello, I'd love to take this as first contribution for this repo. I suggest rename `*Callback` with `*Handler`. WDYT?

I'd prefer to go with `*Func` instead of `*Handler`. This based on the fact that we have an [`ErrorHandler`](https://github.com/open-telemetry/opentelemetry-go/blob/45b6a22cbba2fb3015c50ad095ff61f51b81aece/error_handler.go#L17-L22) that is an interface not fitting the callback-like function form. Using a `*Func` suffix would already have some implementations, be shorter, fit a bit closer to the functional nature of the call (this is subjective for sure), and wouldn't conflict with the existing instance of the `ErrorHandler`.

Additionally, our existing `Bound*` instruments used to be called `Handlers*`. This was changed so there wouldn't be conflict there, but there are still some lurking remnants (maybe just [this](https://github.com/open-telemetry/opentelemetry-go/blob/1cdf4ee8e6181f181af94d7539e033aef47bfc2f/sdk/metric/correct_test.go#L40) so not the biggest deal :shrug:) of this naming that might confuse if we went with `*Handler`.
